### PR TITLE
[wt11] Update docker soft

### PR DESCRIPTION
**Pull request by Sckorn**
GCC: 14
Clang: 18
Bazel: 7.3.0
Vulkan: 1.3.290
Enable C++23
Minor changes and adaptations

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -80,11 +80,11 @@ build:windows --host_copt=-DNOGDI
...
-build:linux_clang16 --host_copt=-Wno-error=implicit-function-declaration
-build:linux_clang16 --copt=-Wno-error=implicit-function-declaration
-build:linux_clang16 --host_copt=-Wno-error=int-conversion
-build:linux_clang16 --copt=-Wno-error=int-conversion
-build:linux_clang16 --config=linux
+build:linux --host_copt=-Wno-error=implicit-function-declaration
+build:linux --copt=-Wno-error=implicit-function-declaration
+build:linux --host_copt=-Wno-error=int-conversion
+build:linux --copt=-Wno-error=int-conversion
+#build:linux_clang16 --config=linux
```
Comment (radameon) [inline .bazelrc:87]: Remove comment

File: docker/docker-compose/radameon_dev.yaml [hunk 2]
```diff
@@ -7,14 +7,20 @@ services:
 
     container_name: radameon_dev
 
+    #environment:
```
Comment (radameon) [inline docker/docker-compose/radameon_dev.yaml:10]: Remove comment

File: docker/docker-compose/radameon_dev.yaml [hunk 3]
```diff
@@ -7,14 +7,20 @@ services:
...
       # tools
       - type: bind
         source: $HOME/.gitconfig
-        target: /home/oper/.gitconfig
+        target: /home/radameon/.gitconfig
 
       - /var/run/docker.sock:/var/run/docker.sock
 
+      
+      #- ~/.Xauthority:/home/radameon/.Xauthority:rw
```
Comment (radameon) [inline docker/docker-compose/radameon_dev.yaml:22]: Remove comment

File: docker/dockerfiles/build.docker [hunk 4]
```diff
@@ -28,14 +28,16 @@ RUN BAZEL_DOWNLOAD_URL="https://github.com/bazelbuild/bazel/releases/download/${
...
     mkdir -p /radameon/web_server/ && \
-    unzip /tmp/web_server.zip -d /radameon/web_server/
+    unzip /tmp/web_server.zip -d /radameon/web_server/ && \
+    mv /radameon/web_server/rcbe_static_storage-fd72a4fa060f369b92517649874899fde682856e/* /radameon/web_server && \
+    rm -r /radameon/web_server/rcbe_static_storage-fd72a4fa060f369b92517649874899fde682856e/
 
 # Install Vulkan SDK
 # Change apt package installation, when it's fixed
-RUN wget -qO- https://packages.lunarg.com/lunarg-signing-key-pub.asc | sudo tee /etc/apt/trusted.gpg.d/lunarg.asc && \
-    wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.231-jammy.list https://packages.lunarg.com/vulkan/1.3.231/lunarg-vulkan-1.3.231-jammy.list && \
```
Comment (radameon) [inline docker/dockerfiles/build.docker:39]: Fix previous version in README.md

File: docker/dockerfiles/build_base.docker [hunk 5]
```diff
@@ -57,8 +54,10 @@ RUN apt-get -qq update && \
...
         libglm-dev \
         libfreetype6-dev \
+        python3-tornado \
         wget
 
 # Install pip packages
 
-RUN pip install --no-cache-dir scipy numpy scikit-learn matplotlib pwlf plyfile protobuf tornado
+#RUN pip install --no-cache-dir scipy numpy scikit-learn matplotlib pwlf plyfile protobuf tornado --break-system-packages 
+#RUN apt install python3-tornado
```
Comment (radameon) [inline docker/dockerfiles/build_base.docker:63]: Remove commented code above

File: docker/scripts/docker_builder.py [hunk 6]
```diff
@@ -74,14 +74,30 @@ def build_file(context, file, image_config, verbose=False):
...
+        success_marker = "Successfully built "
+        suc_string = ret.stdout.split('\n')[-2]
+    else:
+        success_marker = "writing image sha256 "
+        suc_string = ret.stderr.split('\n')[-3]
 
-    success_marker = "Successfully built "
-    suc_string = ret.stdout.split('\n')[-2]
+    #print(suc_string)
+    #print(success_marker)
```
Comment (radameon) [inline docker/scripts/docker_builder.py:86]: Remove comments

File: docker/scripts/docker_builder.py [hunk 7]
```diff
@@ -74,14 +74,30 @@ def build_file(context, file, image_config, verbose=False):
...
+        resulting_digest = suc_string[substr_ind + len(success_marker):-1].strip()
+    else:
+        resulting_digest = suc_string[substr_ind + len(success_marker):-5].split(':')[1].strip()
+
+    if ' ' in resulting_digest:
+        resulting_digest = resulting_digest.split(' ')[0]
+
     print("Resulting digest %s" % resulting_digest)
 
+    #exit(1)
```
Comment (radameon) [inline docker/scripts/docker_builder.py:100]: Remove comment

File: docker/scripts/run_docker [hunk 8]
```diff
@@ -72,4 +72,4 @@ done
 
 # Run container
 echo "Running $service"
-docker-compose -f $yaml_file run --service-ports --rm $service
+docker compose -f $yaml_file run --service-ports --rm $service
```
Comment (radameon) [inline docker/scripts/run_docker:75]: Add xhost command above

**Review by radameon — CHANGES_REQUESTED** [review]
Needs some fixing

File: docker/scripts/run_docker (hunk 8 above)
Comment (Sckorn) [reply docker/scripts/run_docker:76]: Added

File: docker/dockerfiles/build.docker (hunk 4 above)
Comment (Sckorn) [reply docker/dockerfiles/build.docker:39]: Fixed

File: README.md [hunk 9]
```diff
@@ -33,10 +33,23 @@ Compiler and rasterizer support table can be found below.
...
+### Compiler support per revision
+
+|Compiler|Revision|
+|:-------|----------|
+|GCC 14.0.1|master|
+|Clang 18.1.8|master|
+|GCC 13.2|[6101ca2edcc4fbc622791df04afc117911e1c8d6](https://github.com/Sckorn/rcbe_engine/commit/6101ca2edcc4fbc622791df04afc117911e1c8d6)|
+|Clang 16|[6101ca2edcc4fbc622791df04afc117911e1c8d6](https://github.com/Sckorn/rcbe_engine/commit/6101ca2edcc4fbc622791df04afc117911e1c8d6)|
+
 > [!WARNING]
```
Comment (radameon) [inline README.md:53]: Move this warning right below compiler\rasterizer matrix.

**Review by radameon — CHANGES_REQUESTED** [review]
One minor fix, squash, and we're good to go

**Review by radameon — APPROVED** [review]
Merge it once CI is done

**Pull request merged**